### ISSUE TITLE
False positive with unnamed function parameter

### ISSUE THREAD

**Issue by WasabiFan** [cpplint/cpplint#131]
With the following example file:

```c++
void foo(int)
{
    return;
}
```

I get:
```
# cpplint test.cpp 
...
test.cpp:1:  Using C-style cast.  Use static_cast<int>(...) instead  [readability/casting] [4]
...
# cpplint --version
Cpplint fork (https://github.com/cpplint/cpplint)
cpplint 1.4.5
Python 3.8.0 (default, Nov 23 2019, 05:49:00) 
[GCC 8.3.0]
```

There isn't a cast in this file; it's picking up the parameter type in the function definition.

I removed the parameter name to avoid "unused variable" warnings (https://stackoverflow.com/a/15763959/2422874).

**Comment by tkruse** [comment]
It seems that this is avoided when the curly braces are at the end of the line.
cpplint.py already tries to avoid this false positive, but the code to do so in `CheckCasts(` is very contrived, and I am not brave enough to try to fix this.

**Comment by vkucera** [comment]
Any prospects on fixing this bug?
